### PR TITLE
Add task cancellation test

### PR DESCRIPTION
**Pull request by tsoutsman**
It doesn't currently pass, but should in the near future.

Signed-off-by: Klim Tsoutsman <klim@tsoutsman.com>

### CODE REVIEW

File: applications/test_task_cancel/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        unwind_row_generator();
+    }
+}
+
+#[inline(never)]
+fn unwind_row_generator() {
+    static __COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+    // Prevents unwind_row_generator from being optimised away.
+    __COUNTER.fetch_add(1, Ordering::Relaxed);
```
Comment (kevinaboos) [inline applications/test_task_cancel/src/lib.rs]: perhaps `black_box()` would work here too?

**Review by kevinaboos — APPROVED** [review]
Great idea to add this test! I left one small comment but am still approving it; will merge in after discussion.

File: applications/test_task_cancel/src/lib.rs (hunk 1 above)
Comment (tsoutsman) [reply applications/test_task_cancel/src/lib.rs]: https://doc.rust-lang.org/beta/core/hint/fn.black_box.html
> Programs cannot rely on black_box for correctness in any way.

Comment (tsoutsman) [reply applications/test_task_cancel/src/lib.rs]: Either way, we don't need it, as the `FALSE` static prevents the function from being optimised away.

**Pull request merged**